### DISCUSSION TITLE
Parsing XML in background script with import or DOMParser

### DISCUSSION THREAD

**Discussion by MinaroShikuchi** [crxjs/chrome-extension-tools#581] [Q&A]
I have a chrome.alarm that fetch every  1 hour an URL.
The returned content is RSS as xml. 
I would like to parse it with an  external package or with the DOMParser.

I'm using vite and svelte in the vite.config.ts
```js
import { build, defineConfig } from 'vite'
import { svelte } from '@sveltejs/vite-plugin-svelte'
import { crx } from "@crxjs/vite-plugin";
import manifest from './manifest.json' assert { type: 'json' } 

export default defineConfig({
  plugins: [svelte(), crx({ manifest })],
});
```

With an external package I get 
```js
import { parseStringPromise } from "xml2js";

await fetch(url, {
    headers: {
      Accept: "application/xml",
    },
  })
    .then(async (response) => {
      const data = await response.text();
      return await parseStringPromise(data);
    })
    .catch((error) => console.error(error))
```
```
Uncaught TypeError: Cannot read properties of undefined (reading 'prototype')
```
And with DOMParser since DOM is not available in background script
```js

export default function parseXmlToJson(xml: string) {
 return new DOMParser().parseFromString(xml, "text/xml");
}

await fetch(url, {
    headers: {
      Accept: "application/xml",
    },
  })
    .then(async (response) => {
      const data = await response.text();
      return parseXmlToJson(data);
    })
    .catch((error) => console.error(error))
```
```
Error handling response: ReferenceError: DOMParser is not defined
```

Do you have any idea how to achieve this ?

**Comment by jacksteamdev** [comment]
This is a [current limitation](https://bugs.chromium.org/p/chromium/issues/detail?id=1056354&q=component%3APlatform%3EExtensions%20domparser&can=2) of MV3, but the new [Offscreen API](https://bugs.chromium.org/p/chromium/issues/detail?id=1339382) should allow you to do this. I believe it's in canary right now.

I'd probably try using an NPM library that works with service workers.

**Reply by MinaroShikuchi** [reply]
I solved it with some custom regex but it's not ideal

**Comment by honwhy** [comment]
though it is late to answer, but you may try `linkeddom`
https://github.com/WebReflection/linkedom
